workspaces/adventure-pack/goodies/typescript/Iterator.prototype.drop/index.test.ts

```diff
@@ -0,0 +1,78 @@
+import { describe, expect, it } from "@jest/globals";
+
+import { iteratorPrototype } from "../Iterator.prototype";
+delete (iteratorPrototype as unknown as Record<string, unknown>).drop;
+
+import "./index";
+
+describe("Iterator.prototype.drop", () => {
+  it("drops the specified number of elements from the iterator", () => {
+    const result = Array.from([10, 22, 33, 47, 58].values().drop(3));
```

Comment:
Probably a matter of personal taste, but you could use spread syntax instead of `Array.from`.

Also, I am guessing you created all the `result` variables for readability, since they are only used once and could theoretically be inlined. If the goal is readability, what do you think about this style instead?

```js
const iterator = [10, 22, 33, 47, 58].values();

expect([...iterator.drop(3)]).toStrictEqual([47, 58]);
```

Basically we are separating the part where the iterator is created (`.values()` on an array expression, or any other kind of expression we may choose) from the part where the operation being tested is invoked, i.e. `.drop()`. In my opinion that strikes the best readability balance.

Reply:
Sounds good. Readability was indeed my goal, but I find myself struggling at times looking for the right style for making things both readable and concise. I like the above style, glad to adopt it!